lib/src/facts/aix/operating_system_resolver.cc

```diff
@@ -0,0 +1,68 @@
+#include <internal/facts/aix/operating_system_resolver.hpp>
+#include <internal/util/regex.hpp>
+#include <facter/facts/os.hpp>
+#include <facter/execution/execution.hpp>
+#include <facter/util/file.hpp>
+
+#include <sstream>
+#include <algorithm>
+
+using namespace std;
+using namespace facter::util;
+namespace execution = facter::execution;
+
+string getattr(string object, string field)
+{
+    string result;
+
+    // To get architecture, can/should we use getconf KERNEL_BITMODE, getconf HARDWARE_BITMODE?
```

Comment:
I think this is consistent-ish. If the goal of the architecture fact is to
be able to say "what binaries can I drop on this system", I think this is a
reasonable value.

Of course, we don't return feature info (such as SSE support) on x86, so
maybe " power " is enough here
On May 22, 2015 3:49 PM, "Will Hopper" <notifications@github.com> wrote:

> In lib/src/facts/aix/operating_system_resolver.cc
> <https://github.com/puppetlabs/facter/pull/953#discussion_r30936375>:
>
> > +#include <facter/facts/os.hpp>
> > +#include <facter/execution/execution.hpp>
> > +#include <facter/util/file.hpp>
> > +
> > +#include <sstream>
> > +#include <algorithm>
> > +
> > +using namespace std;
> > +using namespace facter::util;
> > +namespace execution = facter::execution;
> > +
> > +string getattr(string object, string field)
> > +{
> > +    string result;
> > +
> > +    // To get architecture, can/should we use getconf KERNEL_BITMODE, getconf HARDWARE_BITMODE?
>
> @MikaelSmith <https://github.com/MikaelSmith> I think the answer is no,
> looking at what lsattr and getconf return:
> getconf KERNEL_BITMODE
> 64
>
> getconf HARDWARE_BITMODe
> 64
>
> lsattr -El proc0 -a type
> type PowerPC_POWER7 Processor type False
>
> lsattr -El sys0 -a modelname
> modelname IBM,8231-E1D Machine name False
>
> Although I can't entirely recall how we're determining the actual
> architecture with lsattr. If we're reporting POWERPC..., it isn't
> consistent with other platforms that just return x86_64 or i386
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/facter/pull/953/files#r30936375>.
>